### DISCUSSION TITLE
Wikijs Large File download and upload  issuer

### DISCUSSION THREAD

**Discussion by iranadu** [requarks/wiki#3591] [Error / Bug Report]
**Describe the bug**
if I try to download the files such as more than 500GB it gets an error. it means it not downloaded. if I used a reverse proxy server such as Apache or Nginx it gets 503 error. how can I resolve this problem? and also max file uploads size only 1 GB. if I change the parameters of the config file. it, not workes. how can I handle those things?   

**To Reproduce**
Steps to reproduce the behaviour:
1. Upload a file 'ore than 100 MB'
2. then try to download 

**Host Info (please complete the following information):**
 - OS: [CentOs 7.8  ]
 - Wiki.js version: [2.4.107 Latest ]
 - Database engine: [mysql 8.0]

**Comment by Smankusors** [comment]
if Apache or Nginx is returning 503, it probably timeout on reverse proxy side, not Wiki.js.

You might need to increase timeout on Apache/Nginx.

**Comment by NGPixel** [comment]
You shouldn't store large files (especially not 500GB files) in Wiki.js. It's a document storage (PDFs, images, etc.), not a large file storage platform. Use something like AWS S3 or Azure Blob Storage instead.

Proper large file support will be added in a future release.

**Reply by mjamro** [reply]
The problem is the default value of 5MB is just ridicoulosly low.

**Comment by iranadu** [comment]
@NGPixel Then I try to download a file, I **Can not** download more than **25MB** Files. 

We have already increased Apache/Nginx time out limit.

**Comment by Smankusors** [comment]
> @NGPixel Then I try to download a file, I Can not download more than **25MB** Files.
> 
> We have already increased Apache/Nginx time out limit.

wait, download or upload?

And why you said Apache/Nginx? You used them both together? Or you used it one at the time? 😕

**Comment by iranadu** [comment]
@NGPixel @Smankusors The issue for downloading an uploaded file 

We tried both web servers but we couldn't download  More than **25 MB** file 

We already expand the upload size of wikijs.config XML file up to 1 GB.

**Comment by NGPixel** [comment]
Have you tried **without** nginx / apache in front first? If you get the same issue downloading a file directly from wiki.js, then we can investigate further.

**Comment by iranadu** [comment]
@NGPixel Thanks for your reply Yes, we tried to access it directly with IP:3000 but the same issue, 
We couldn't find any configuration file to solve this. is there another config file changes to fix this.

**Comment by Smankusors** [comment]
> @NGPixel Thanks for your reply Yes, we tried to access it directly with IP:3000 but the same issue,
> We couldn't find any configuration file to solve this. is there another config file changes to fix this.

what do you mean by same issue? Did you get 503 like you first stated or timeout error or something? 🤔

**Comment by iranadu** [comment]
@Smankusors  if I **use** Apache or Nginx web server via download more than **25MB** file, I got 503 error ( i already increased timeout limit)

if I use **without** Apache or Nginx webserver ( access directly)  I **can not** download more than **25MB**  file.

**Comment by KingRial** [comment]
I am testing the same situation right now.
I can freely handle files bigger than 25Mb but I am getting a different error.

Here what I did:

- Cloned the repository
- Switched to stable version 2.4.107
- Launched development environment by using Visual Studio Code, Remote Containers and the command "yarn dev"
- Set 500Mb as max file size under "Administration -> Security -> Uploads"

I can handle, without problems, uploads/downloads up to about 100Mb

When I reach 200Mb the "yarn dev" command explodes with the following error:
![image](https://user-images.githubusercontent.com/886518/92395743-1dd77e80-f124-11ea-9126-335d94a4893b.png)

Problably linked to "container out of memory" problems

**Comment by dongshimou** [comment]
same error.
docker image [latest](https://github.com/Requarks/wiki/blob/2.5.117/dev/build/Dockerfile)
can't not download large file. 500 Internal Server Error.
25mb exe file. ok
28mb pdf file. ok
42mb exe file. failed
54mb exe file. failed

**Comment by 01beacon** [comment]
Same error with files over 10mb on RHEL8 and Apache. Tried both without reverse proxy and the file size is increased. Can upload but not play or download files.

**Reply by maxim43g** [reply]
Are you planning to fix the bug?
I can't download files larger than 10 megabytes. I am uploading without problems.
